x/concentrated-liquidity/incentives_test.go

```diff
@@ -3509,3 +3511,241 @@ func (s *KeeperTestSuite) TestPrepareBalancerPoolAsFullRange() {
+
+			// --- Assertions ---
+
+			if tc.expectedError != nil {
+				s.Require().Error(err)
+				s.Require().ErrorContains(err, tc.expectedError.Error())
+				s.Require().Equal(sdk.Coins{}, amountClaimed)
+
+				clPoolUptimeAccumulators, err := s.App.ConcentratedLiquidityKeeper.GetUptimeAccumulators(s.Ctx, clPool.GetId())
+				s.Require().NoError(err)
+
+				for _, uptimeAccum := range clPoolUptimeAccumulators {
+					currAccumShares, err := uptimeAccum.GetTotalShares()
+					s.Require().NoError(err)
+
+					// Ensure accum shares remain unchanged after error
+					s.Require().Equal(initialLiquidity.Add(addedLiquidity), currAccumShares)
+				}
+
+				// If gauge exists, ensure it remains empty after error
+				if !tc.balancerPoolDoesNotExist {
+					gaugeId, err := s.App.PoolIncentivesKeeper.GetPoolGaugeId(s.Ctx, balancerPoolId, longestLockableDuration)
+					s.Require().NoError(err)
+
+					gauge, err := s.App.IncentivesKeeper.GetGaugeByID(s.Ctx, gaugeId)
+					s.Require().NoError(err)
+
+					s.Require().Equal(sdk.Coins(nil), gauge.Coins)
+				}
+
+				// Pool liquidity should remain unchanged
+				updatedClPool, err := s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, clPool.GetId())
+				s.Require().NoError(err)
+				s.Require().Equal(initialLiquidity, updatedClPool.GetLiquidity())
+
+				return
+			}
+
+			s.Require().NoError(err)
+
+			clPoolUptimeAccumulators, err := s.App.ConcentratedLiquidityKeeper.GetUptimeAccumulators(s.Ctx, clPool.GetId())
+			s.Require().NoError(err)
+
+			for _, uptimeAccum := range clPoolUptimeAccumulators {
+				currAccumShares, err := uptimeAccum.GetTotalShares()
+				s.Require().NoError(err)
+
+				// Ensure each accum has been cleared of the balancer full range shares
+				s.Require().Equal(initialLiquidity, currAccumShares)
+			}
+
+			// Get balancer gauge corresponding to the longest lockable duration, as this is the one we would be distributing to
+			gaugeId, err := s.App.PoolIncentivesKeeper.GetPoolGaugeId(s.Ctx, balancerPoolId, longestLockableDuration)
+			s.Require().NoError(err)
+			gauge, err := s.App.IncentivesKeeper.GetGaugeByID(s.Ctx, gaugeId)
+			s.Require().NoError(err)
+
+			// Since balancer position bonding is a stronger constraint than CL charging, we never need to forfeit incentives
+			longestSupportedUptime := types.SupportedUptimes[len(types.SupportedUptimes)-1]
+
+			// Calculate the number of tokens we expect to see in the balancer gauge
+			expectedTokensInGauge := expectedIncentivesFromUptimeGrowth(tc.uptimeGrowth, addedLiquidity, longestSupportedUptime, defaultMultiplier)
+
+			s.Require().Equal(expectedTokensInGauge, gauge.Coins)
+
+			// Pool liquidity should remain unchanged
+			updatedClPool, err := s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, clPool.GetId())
+			s.Require().NoError(err)
+			s.Require().Equal(initialLiquidity, updatedClPool.GetLiquidity())
```

Comment:
Is there an assertion that uptime accumulators for full range balancer position are emptied? If not, please add it prior to merge

Reply:
I'm assuming you mean that the full range records have no more shares and are cleared from state. We check the former above in the following section:
```
// Ensure each accum has been cleared of the balancer full range shares
s.Require().Equal(initialLiquidity, currAccumShares)
```

Will add a check for the latter